### PR TITLE
Remove complex numbers from the Getkw grammar

### PR DESCRIPTION
**Pull request by robertodr**
This adheres to the original "standard" which did not have complex numbers as available tokens.

### CODE REVIEW

**Comment by bast** [comment]
Why is it important to adhere to the original standard? I believe it is but don't yet know why.

File: parselglossy/read_yaml.py [hunk 1]
```diff
@@ -1,11 +1,39 @@
...
+# For information on the complete list of contributors to the
+# parselglossy library, see: <http://parselglossy.readthedocs.io/>
+#
+
+from pathlib import Path
 
+import yaml
 
-JSONDict = Dict[str, Any]
+from .utils import JSONDict
```
Comment (bast) [inline parselglossy/read_yaml.py:33]: very good to collect these in a file - perhaps we should call the file `types` - but we can do this later.

File: parselglossy/read_yaml.py [hunk 2]
```diff
@@ -1,11 +1,39 @@
...
 
 
-def read_yaml_file(file_name: str) -> JSONDict:
+def read_yaml_file(file_name: Path) -> JSONDict:
```
Comment (bast) [inline parselglossy/read_yaml.py:36]: very good - thanks!

File: tests/test_validation.py [hunk 3]
```diff
@@ -1,21 +1,43 @@
...
+from parselglossy.utils import JSONDict
+from parselglossy.validate import (InputError, TemplateError, check_predicates_node, validate_node)
 
 
-def _helper(category: str,
-            input_file_name: str,
-            template_file_name: str) -> JSONDict:
-    this_path = Path(os.path.dirname(os.path.realpath(__file__)))
+def _helper(category: str, input_file_name: str, template_file_name: str) -> JSONDict:
+    this_path = Path(__file__).parent
```
Comment (bast) [inline tests/test_validation.py:40]: nice!

**Review by bast — APPROVED** [review]

**Comment by robertodr** [comment]
Just so that there no surprises for those who think this is an exact replacement.

**Comment by bast** [comment]
But are we aiming at an exact replacement? It's a different name, different version and I think the API will also be different. But my questions here are not a blocker to this PR.

**Comment by robertodr** [comment]
It will never be an exact replacement, because validation is now completely separate from the grammar definition. However, I think it's cleaner to offer the option of _not_ having complex numbers in the grammar.

**Pull request merged**